mindsdb/libs/phases/stats_generator/data_preparation.py

```diff
@@ -0,0 +1,36 @@
+import random
+import numpy as np
+
+from mindsdb.external_libs.stats import calculate_sample_size
+from mindsdb.libs.constants.mindsdb import *
+from mindsdb.libs.helpers.text_helpers import clean_float
+
+def sample_data(df, sample_margin_of_error, sample_confidence_level, log):
+    population_size = len(df)
+
+    sample_size = int(calculate_sample_size(population_size, sample_margin_of_error, sample_confidence_level)) if population_size > 50 else population_size
+    sample_size_pct = sample_size*100/population_size
+
+    # get the indexes of randomly selected rows given the population size
+    input_data_sample_indexes = random.sample(range(population_size), sample_size)
+
+    log.info(f'Analyzing a sample of {sample_size} from a total population of {population_size}, this is equivalent to {sample_size_pct}% of your data.')
+
+    return df.iloc[input_data_sample_indexes]
+
+
+def clean_int_and_date_data(col_data, log):
+    cleaned_data = []
+
+    for ele in col_data:
+        if str(ele) not in ['', str(None), str(False), str(np.nan), 'NaN', 'nan', 'NA', 'null'] and ele != '' and ele != '\r' and ele != '\n':
```

Comment:
We can use `and (not ele.isspace())` that will check for space, tab, newline, carriage return and feed'\f'